### ISSUE TITLE
trigsimp doesn't simplify completely

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#18231]
`trigsimp((cos(x-pi*cos(x))-cos(x+pi*cos(x)))*cos(x))` gives `2sin(x)cos(x)sin(pi*cos(x))` and `trigsimp(trigsimp((cos(x-pi*cos(x))-cos(x+pi*cos(x)))*cos(x)))` gives `sin(2x)sin(pi*cos(x))`

**Comment by gschintgen** [comment]
What Sympy version are you using? I can't reproduce the double trigsimp invocation:
```
In [1]: trigsimp(2*sin(x)*cos(x)*sin(pi*cos(x)))
Out[1]: 2⋅sin(x)⋅sin(π⋅cos(x))⋅cos(x)

In [2]: trigsimp(trigsimp(2*sin(x)*cos(x)*sin(pi*cos(x))))
Out[2]: 2⋅sin(x)⋅sin(π⋅cos(x))⋅cos(x)
```
Would you mind copy & pasting the *exact* input and output?

`simplify` also doesn't contract the `2*sin(x)*cos(x)` factors. Neither does `trigsimp(... method='fu')`.

(That's with current master.)

**Comment by proy87** [comment]
Indeed, double `trigsimp` doesn't simplify. Anyway, the problem remains.
